pkg/user/user_integration_test.go

```diff
@@ -206,6 +208,42 @@ func TestUserHandler(t *testing.T) {
 				client.Do(t, http.MethodDelete, "/users/"+user1ID, nil, http.StatusUnauthorized, inttest.WithAuthToken(user2Token.AccessToken))
 			}
 		})
+
+		t.Run("ResetUserPassword", func(t *testing.T) {
+			t.Parallel()
+
+			{
+				t.Log("RequestPasswordReset")
+
+				requestResetRequestBody := strings.NewReader(`{
+					"email":    "user1@dhis2.org"
+				}`)
+
+				client.Do(t, http.MethodPost, "/users/request-reset", requestResetRequestBody, http.StatusOK, inttest.WithHeader("Content-Type", "application/json"))
```

Comment:
Don't we have a client.PostJSON (or such) method?

Reply:
We do, but [it expects a 201 response](https://github.com/dhis2-sre/im-manager/blob/03422fb6e388b7bf8dca7fdb7d10a6348420b399/pkg/inttest/http.go#L104), while we get a 200 here.

This begs the question - should we use a 201 response for both of the request password reset and reset password endpoints? 🤔

Reply:
I would be ok with returning 201. But I'll leave it up to you how we proceed here :slightly_smiling_face:

Reply:
I suppose it makes sense to return 201, as we are creating a password reset token and then updating the password. 🤔

Reply:
I actually also thought a bit more about it, and yes, 201 is the way to go :+1: